### DISCUSSION TITLE
Apple Silicon Streaming

### DISCUSSION THREAD

**Discussion by ryansch** [LizardByte/community#440] [Sunshine] [answered]
Some googling suggests that OBS can use the hardware encoder on Apple Silicon machines.

What would it look like to support Sunshine on these devices for game streaming?

**Comment by ReenigneArcher** [comment, marked as answer]
https://github.com/LizardByte/Sunshine/pull/2485

**Comment by Hazer** [comment]
> #2485

It may be related, as the fix only changes how long the frame is available, so I guess if it's the same issue, this should solve it there too, but I don't have Apple Silicon to try it out.

@ryansch if you use homebrew and want to try out my fix, I'd love to hear from you, you can take the formula file generated for the PR here:
https://github.com/LizardByte/Sunshine/actions/runs/8897937352/artifacts/1461467634
unzip the downloaded file, then in the same folder you unzipped, run to install:
`brew install --build-from-source sunshine.rb`

It's a bit slower to install because it's building from source, but should work.

**Reply by ReenigneArcher** [reply]
> It's a bit slower to install because it's building from source, but should work.

I think our brew formula does that anyway. Is there a way to provide it pre-built?

**Reply by Hazer** [reply]
> I think our brew formula does that anyway. Is there a way to provide it pre-built?

I see, if I'm not mistaken it's possible to do using brew bottles, but I never did it myself, I never really "brewed" any brew.

**Reply by Hazer** [reply]
Just updating, now that PR actually includes a fix for hardware encoders not being recognized, the fix should be for any macOS. But I don't have an Apple Silicon to test.

**Comment by Hazer** [comment]
@ryansch should be fixed on nightly, some Apple Silicon users reported with good results.
